File: 0014 最长公共前缀/main.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        nl = 0
        flag = True
        LCP = ""
        if(len(strs) == 0):
            return LCP
        if(len(strs) == 1):
            return strs[0]
        for i in strs:
            if(i == ""):
                return ""
        while flag:
            if(nl == len(strs[0])):
                break
            tmp = strs[0][nl]
            for i in strs:
                if(nl == len(i)):
                    flag = False
                    break
                if(tmp != i[nl]):
                    flag = False
                    break
            if(flag == True):
                LCP += tmp
                nl += 1
        return LCP

File: 0014 最长公共前缀/main.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==0){
            return "";
        }
        string res=strs[0];
        string s;
        for(int i=1;i<strs.size();i++){
            s=strs[i];
            int j=0;
            for(;j < res.size();j++){
                if( j >= s.size() || s[j] != res[j]){
                    break;
                }
            }
            res=res.substr(0,j);
            if(res == ""){
                return "";
            }
        }
        return res;
    }
};